fastify.js

```diff
@@ -662,6 +663,30 @@ function fastify (options) {
     res.end(body)
   }
 
+  function buildAsyncConstraintCallback (isAsync, req, res) {
+    if (isAsync === false) return undefined
+    return function onAsyncConstraintError (err) {
+      if (err) {
+        if (frameworkErrors) {
+          const id = genReqId(req)
+          const childLogger = logger.child({ reqId: id })
+
+          childLogger.info({ req }, 'incoming request')
+
+          const request = new Request(id, null, req, null, childLogger, onBadUrlContext)
+          const reply = new Reply(res, request, childLogger)
+          return frameworkErrors(new FST_ERR_ASYNC_CONSTRAINT(), request, reply)
+        }
+        const body = '{"error":"Internal Server Error","message":"Unexpected error from async constraint","statusCode":500}'
+        res.writeHead(500, {
+          'Content-Type': 'application/json',
+          'Content-Length': body.length
+        })
+        res.end(body)
```

Comment:
Did you evaluate `res.destroy` instead?

The only difference is whether or not emit an `error` from the request object. Can't recall any other differences

Reply:
Not quite understand what you means.

Reply:
I needed to refresh this

```
const { fastify } = require('fastify')
const app = fastify()

app.get('/destroy', async (request, reply) => {
  reply.hijack()
  reply.raw.destroy('hello world')
})

app.get('/end', async (request, reply) => {
  reply.hijack()
  reply.raw.end('hello world')
})

app.inject('/end', check)
  .then(() => app.inject('/destroy', check))

function check (err, res) {
  console.log(err)
  console.log(res?.statusCode)
  console.log(res?.payload)
}
```

Reply:
`.destory` cannot and will not send any payload to the client. Instead, it close the socket immediately.
`.end` return a proper error message to the client and they can deal with it other than just wait for browser socket timeout.